### PR TITLE
[NT-864] Continue with apple UI

### PR DESCRIPTION
**Pull request by Scollaco**
# 📲 What

- New SignUp screen with the `Continue with Apple` button for iOS 13+ users

# 🤔 Why
[JIRA ticket](https://kickstarter.atlassian.net/browse/NT-864)

- This is part of the work that implements the new signup flow required by apple.

# 🛠 How

- Added a `ASAuthorizationAppleIDButton` to the view hierarchy
- Added availability checks to hide the `Continue with Apple` button if the app is running on iOS 12
- Added a new background image
- Generated new Snapshots

# 👀 See

| Before 🐛 | After 🦋iOS 12 | After 🦋iOS 13+ |
| --- | --- | --- |
|  |  |  |

**Note:** After talking to Jimmy on Slack, we decided to keep the space between the Facebook and Signup buttons.

- [ ] Tap targets use minimum of 44x44 pts dimensions
- [ ] Works with VoiceOver

# ✅ Acceptance criteria

- [x] Running the app on iOS 13, the `Cotinue with Apple` button should be visible on the Login screen. It's hidden on iOS 12.

### CODE REVIEW

File: Kickstarter-iOS/Views/Controllers/LoginToutViewController.swift [hunk 1]
```diff
@@ -85,6 +88,11 @@ internal final class LoginToutViewController: UIViewController, MFMailComposeVie
       self.navigationItem.leftBarButtonItem = .close(self, selector: #selector(self.closeButtonPressed))
     }
 
+    _ = self.navigationController?.navigationBar
+      ?|> \.barTintColor .~ .clear
```
Comment (ifbarrera) [inline Kickstarter-iOS/Views/Controllers/LoginToutViewController.swift]: I created a helper function `configureTransparentNavigationBar()` as an extension of `UINavigationBar` that you can use to set the navigation bar to be transparent. It's currently in the `feature-onboarding-personalization` branch, but maybe make a note here to update this once that's merged?

File: Kickstarter-iOS/Views/Controllers/LoginToutViewController.swift [hunk 2]
```diff
@@ -440,7 +486,20 @@ private let baseStackViewStyle: StackViewStyle = { stackView in
...
+
+private let logoImageViewStyle: ImageViewStyle = { imageView in
+  imageView
+    |> \.image .~ image(named: "kickstarter-logo")?.withRenderingMode(.alwaysTemplate)
+    |> \.tintColor .~ .ksr_green_500
+    |> \.contentMode .~ .scaleAspectFit
+    |> \.translatesAutoresizingMaskIntoConstraints .~ false
+}
+
+private let separatorViewStyle: ViewStyle = { view in
```
Comment (ifbarrera) [inline Kickstarter-iOS/Views/Controllers/LoginToutViewController.swift]: We already have `separatorStyle` in `BaseStyles.swift`. We should try to avoid having many different styles for the same component, would the one we already have work?

File: Library/ViewModels/LoginToutViewModel.swift [hunk 3]
```diff
@@ -58,6 +55,9 @@ public protocol LoginToutViewModelOutputs {
   /// Emits an access token envelope that can be used to update the environment.
   var logIntoEnvironment: Signal<AccessTokenEnvelope, Never> { get }
 
+  /// Emits an image to be used as the navigation bar background.
+  var navigationBarBackgroundImage: Signal<UIImage?, Never> { get }
```
Comment (ifbarrera) [inline Library/ViewModels/LoginToutViewModel.swift]: Why is this needed? I noticed when testing that when scrolling, the navigation bar cuts off the scroll content view. Is that the desired behavior or do we want the navigation bar to fully transparent?

Screen Shot 2020-03-30 at 3 03 49 PM

File: Kickstarter-iOS/Views/Controllers/LoginToutViewController.swift (hunk 1 above)
Comment (Scollaco) [reply Kickstarter-iOS/Views/Controllers/LoginToutViewController.swift]: Oh good to know! I wonder if I could cherry-pick the commit

File: Library/ViewModels/LoginToutViewModel.swift (hunk 3 above)
Comment (Scollaco) [reply Library/ViewModels/LoginToutViewModel.swift]: The `navigationBarBackgroundImage` to change color to `white` when going to signup and login screens (view with white background). I think that using your extension should fix this and this output shouldn't be necessary anymore.

File: Kickstarter-iOS/Views/Controllers/LoginToutViewController.swift (hunk 1 above)
Comment (Scollaco) [reply Kickstarter-iOS/Views/Controllers/LoginToutViewController.swift]: Actually I copied and pasted the extension code and this is what I got:



Another solution would be to hide the navigation bar and add the buttons on the top corners. What do you think?

Comment (ifbarrera) [reply Kickstarter-iOS/Views/Controllers/LoginToutViewController.swift]: @nino I think you need to remove the `baseControllerStyle` in order for it to work. `baseControllerStyle` is applying it's own navigation bar style that's overriding our setup.

Comment (nino) [reply Kickstarter-iOS/Views/Controllers/LoginToutViewController.swift:92]: (Just FYI I don't think you meant to write my name. (I mean, I know _some_ Swift, but I really couldn't tell you if removing the `baseControllerStyle` would help))

**Comment by nativeksr** [comment]
### SwiftFormat found issues:

| File | Rules |
| ---- | ----- |
| Kickstarter-iOS/Views/Controllers/LoginToutViewController.swift | indent |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by nativeksr** [comment]
### SwiftFormat found issues:

| File | Rules |
| ---- | ----- |
| Kickstarter-iOS/Views/Controllers/LoginToutViewController.swift | indent |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Kickstarter-iOS/Views/Controllers/LoginToutViewControllerTests.swift [hunk 4]
```diff
@@ -33,6 +33,7 @@ internal final class LoginToutViewControllerTests: TestCase {
     let intent = LoginIntent.generic
     let controller = LoginToutViewController.configuredWith(loginIntent: intent)
 
-    XCTAssertNotNil(controller.view.subviews.first as? UIScrollView)
+    // Due to the new design, a background imageView is now added as subview before the scroll.
+    XCTAssertNotNil(controller.view.subviews[1] as? UIScrollView)
```
Comment (ifbarrera) [inline Kickstarter-iOS/Views/Controllers/LoginToutViewControllerTests.swift:37]: Hmm what's this test for? Does this change mean that the scroll to top functionality won't work for this screen?

Comment (Scollaco) [reply Kickstarter-iOS/Views/Controllers/LoginToutViewControllerTests.swift:37]: Correct. The problem here is that in order to match the designs, the view hierarchy had to change, since now a imageView was added to stack before the scrollView. As an option, I tried adding the background image view to the scrollView instead, but the scrolling looks weird with the background image moving.

Comment (Scollaco) [reply Kickstarter-iOS/Views/Controllers/LoginToutViewControllerTests.swift:37]: Oh one sec. I think I found a solution.
